File: PaymentSDK/UIHostObjCApp/sdk/ApplePaySDKDelegate.m
//
//  ApplePaySDKDelegate.m
//  UIHostObjCApp
//
//  Created by Rahul Dhuri on 30/05/19.
//  Copyright © 2019 Network International. All rights reserved.
//

#import "ApplePaySDKDelegate.h"
#import <PassKit/PassKit.h>
@implementation ApplePaySDKDelegate

- (void)applePayContactUpdatedWithDidSelect:(PKContact * _Nonnull)shippingContact handler:(void (^ _Nonnull)(PKPaymentRequestShippingContactUpdate * _Nonnull))completion {
    NSLog(@"Apple pay contact updated: \(shippingContact)");
    completion([[PKPaymentRequestShippingContactUpdate alloc] initWithErrors: nil paymentSummaryItems: [NSArray new] shippingMethods: [NSArray new]]);
}

- (void)applePayPaymentMethodUpdatedWithDidSelect:(PaymentMethod * _Nonnull)paymentMethod handler:(void (^ _Nonnull)(PKPaymentRequestPaymentMethodUpdate * _Nonnull))completion {
    NSLog(@"Apple pay payment method updated: \(paymentMethod)");
    completion([[PKPaymentRequestPaymentMethodUpdate alloc] initWithPaymentSummaryItems: [NSArray new]]);
}

- (void)applePayShippingMethodUpdatedWithDidSelect:(PKShippingMethod * _Nonnull)shippingMethod handler:(void (^ _Nonnull)(PKPaymentRequestShippingMethodUpdate * _Nonnull))completion {
    NSLog(@"Apple pay shipping method updated: \(shippingMethod)");
    completion([[PKPaymentRequestShippingMethodUpdate alloc] initWithPaymentSummaryItems: [NSArray new]]);
}

@end

File: PaymentSDK/UIHostApp/sdk/ApplePaySDKDelegate.swift
//
//  ApplePaySDKDelegate.swift
//  UIHostApp
//
//  Created by Niraj Chauhan on 5/5/19.
//  Copyright © 2019 Network International. All rights reserved.
//

import Foundation
import PassKit
import PaymentSDK

final class ApplePaySDKDelegate : ApplePayDelegate{
    func applePayPaymentMethodUpdated(didSelect paymentMethod: PaymentMethod, handler completion: @escaping (PKPaymentRequestPaymentMethodUpdate) -> Void) {
        print("Apple pay payment method updated: \(paymentMethod)")
        completion(PKPaymentRequestPaymentMethodUpdate(paymentSummaryItems: []))
    }
    
    func applePayShippingMethodUpdated(didSelect shippingMethod: PKShippingMethod, handler completion: @escaping (PKPaymentRequestShippingMethodUpdate) -> Void) {
        print("Apple pay shipping method updated: \(shippingMethod)")
        completion(PKPaymentRequestShippingMethodUpdate(paymentSummaryItems: []))
    }
    
    func applePayContactUpdated(didSelect shippingContact: PKContact, handler completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void) {
        print("Apple pay contact updated: \(shippingContact)")
        completion(PKPaymentRequestShippingContactUpdate(errors: nil,
                                                         paymentSummaryItems: [],
                                                         shippingMethods: []))
    }
    
    
}
